extensions-contrib/kubernetes-overlord-extensions/src/main/java/org/apache/druid/k8s/overlord/KubernetesPeonLifecycle.java

```diff
@@ -136,14 +141,16 @@ protected synchronized TaskStatus run(Job job, long launchTimeout, long timeout,
           launchTimeout,
           TimeUnit.MILLISECONDS
       );
-
       return join(timeout);
     }
     catch (Exception e) {
       log.info("Failed to run task: %s", taskId.getOriginalTaskId());
       throw e;
     }
     finally {
+      if (!taskStartedSuccessfullyFuture.isDone()) {
```

Comment:
taskStartedSuccessfullyFuture is not set to `true` in `run` as far as I can tell. is that intentional?

Reply:
run always calls into join (line 140) so it'll get set there

Reply:
ah, I see, OK.